core/src/main/java/feign/RequestTemplate.java

```diff
@@ -801,7 +824,7 @@ private RequestTemplate appendHeader(String name, Iterable<String> values, boole
       this.headers.remove(name);
       return this;
     }
-    if (name.equals("Content-Type")) {
+    if (name.equalsIgnoreCase("Content-Type")) { // headers are case-insensitive
```

Comment:
Generally better to swap this comparison

```suggestion
    if (("Content-Type").equalsIgnoreCase(name)) { // headers are case-insensitive
```

Reply:
Not sure why I did it backwards - thanks for catching that.